lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -334,6 +334,17 @@ def gem_version_compare(first, second)
             Gem::Version.new(first.to_s) <=> Gem::Version.new(second.to_s)
           end
 
+          apipie :method, 'Returns content of Foreman SSL certificate' do
+            example 'foreman_ssl_certificate # => Content of file'
+          end
+          def foreman_ssl_certificate
+            if File. exist?(Setting[:ssl_ca_file])
+              File.read(Setting[:ssl_ca_file])
+            else
+              raise Foreman::Exception.new("Foreman SSL certificate '#{Setting[:ssl_ca_file]}' not found")
```

Comment:
Are exceptions a good idea in macros? You can't catch them. Also note that you do not need to run Foreman using SSL (like in development setups).

`ssl_ca_file` is listed as a blank attr so it's a valid scenario to encounter. `File.exist?('')` returns `false` and you'll end up here in your development setup.

Returning an empty string is actually fine IMHO.

Reply:
I believe we correctly rescue from exceptions during rendering and I'm sure we use them https://github.com/theforeman/foreman/blob/develop/lib/foreman/renderer/errors.rb

There was already a discussion on this, we agreed to upgrade the communication to https at this stage. We may reconsider and start supporting HTTP in future, if we see production setups, where it's demanded. But dev setup does not qualify for keeping unsecure communication I think.

Reply:
I'd be OK with only supporting that, but then I think we should improve the error reporting to make it actionable for the user. `Foreman SSL certificate` is again a bad description. As a user I would not connect this to the actual setting since the descriptions do not match.

Perhaps it also makes sense to special case the blank value and raise an exception for that? `Foreman SSL certificate '' not found` may cause frustration while `No SSL CA file is configured. Set the `ssl_ca_file` setting` (or similar) is more instructive in how to solve it as a user.